### PR TITLE
Inherit node value and modelling rule of intermediate types

### PR DESCRIPTION
**Pull request by Pro**
See #1363 for a more detailed description

### CODE REVIEW

**Comment by jpfr** [comment]
There's a lot going on in the first commit.
Can we separate cosmetic fixes and functional changes?

**Comment by Pro** [comment]
The cosmetic change is also part of the #1387 so i will first merge that and then this one will be cleaner

**Comment by Pro** [comment]
@jpfr ok this is now rebased on master.
Feel free to review/merge as soon as all green.

File: src/server/ua_nodes.c [hunk 1]
```diff
@@ -575,7 +575,7 @@ void UA_Node_deleteReferences(UA_Node *node) {
         UA_NodeReferenceKind *refs = &node->references[i];
         for(size_t j = 0; j < refs->targetIdsSize; ++j)
             UA_ExpandedNodeId_deleteMembers(&refs->targetIds[j]);
-        UA_free(refs->targetIds);
+        UA_Array_delete(refs->targetIds, refs->targetIdsSize, &UA_TYPES[UA_TYPES_EXPANDEDNODEID]);
```
Comment (jpfr) [inline src/server/ua_nodes.c]: That's cool! Then we don't need the for-loop just above.

File: src/server/ua_services_nodemanagement.c [hunk 2]
```diff
@@ -335,6 +335,77 @@ static void
 addReference(UA_Server *server, UA_Session *session,
              const UA_AddReferencesItem *item, UA_StatusCode *retval);
 
+
+/*
+ * This method only deletes references from the node which are not mathing any type in the given array.
```
Comment (jpfr) [inline src/server/ua_services_nodemanagement.c]: mathing typo.

File: src/server/ua_services_nodemanagement.c [hunk 3]
```diff
@@ -335,6 +335,77 @@ static void
...
+        return;
+    }
+
+    /* Now copy the remaining references to a new array */
+    UA_NodeReferenceKind *newReferences = (UA_NodeReferenceKind *)UA_malloc(sizeof(UA_NodeReferenceKind) * (newSize));
+    size_t curr = 0;
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+    for(size_t i = 0; i < node->referencesSize && retval == UA_STATUSCODE_GOOD; ++i) {
+        for (size_t j = 0; j < referencesSkipSize; j++) {
+            if (UA_NodeId_equal(&node->references[i].referenceTypeId, &referencesSkip[j])) {
```
Comment (jpfr) [inline src/server/ua_services_nodemanagement.c]:
```c
if(!nodeid_equal(...))
    continue;
```

makes the code in the inner loop easier. One level of indendation less.

**Pull request merged**